pkg/webhooks/generation.go

```diff
@@ -50,15 +50,37 @@ func (ws *WebhookServer) HandleGenerate(request *v1beta1.AdmissionRequest, polic
 	for _, policy := range policies {
 		policyContext.Policy = *policy
 		engineResponse := engine.Generate(policyContext)
+		for i, rule := range engineResponse.PolicyResponse.Rules {
+			if !rule.Success {
+				grList, err := ws.kyvernoClient.KyvernoV1().GenerateRequests(config.KubePolicyNamespace).List(metav1.ListOptions{})
+				if err != nil {
+					logger.Error(err, "failed to list generate request")
+				}
+				for _, v := range grList.Items {
+					if engineResponse.PolicyResponse.Policy == v.Spec.Policy && engineResponse.PolicyResponse.Resource.Name == v.Spec.Resource.Name && engineResponse.PolicyResponse.Resource.Kind == v.Spec.Resource.Kind && engineResponse.PolicyResponse.Resource.Namespace == v.Spec.Resource.Namespace {
+						err := ws.kyvernoClient.KyvernoV1().GenerateRequests(config.KubePolicyNamespace).Delete(v.GetName(),&metav1.DeleteOptions{})
+						if err != nil {
+							logger.Error(err, "failed to update gr")
+						}
+					}
+				}
+				if len(engineResponse.PolicyResponse.Rules) > 1 {
+					engineResponse.PolicyResponse.Rules = append(engineResponse.PolicyResponse.Rules[:i], engineResponse.PolicyResponse.Rules[i+1:]...)
+					continue
+				}else if len(engineResponse.PolicyResponse.Rules) == 1 {
+					engineResponse.PolicyResponse.Rules = []response.RuleResponse{}
+				}
+			}
+		}
```

Comment:
Duplicated code of what's in `pkg/generate/generate.go`,  webhook calls `engine.Generate`, do we really need this? Can we remove one of them?

Reply:
yes because here also we are using engine.Generate() to create engineResponse